grc/core/blocks/block.py

```diff
@@ -539,6 +540,30 @@ def is_virtual_sink(self):
     def is_virtual_source(self):
         return self.key == 'virtual_source'
 
+    @property
+    def deprecated(self):
+        """
+        Check whether the block is deprecated.
+
+        For now, we just check the category name for presence of "deprecated".
+
+        As it might be desirable in the future to have such "tags" be stored
+        explicitly, we're taking the detour of introducing a property.
+        """
+        if not self.category:
+            return False
+
+        try:
+            if type(self.category) is list:
```

Comment:
Using `isinstance()` is usually preferred. However, when is self.category not a list?https://github.com/gnuradio/gnuradio/blob/39a173fe78a127e57019600e52173e02e492827d/grc/core/blocks/_build.py#L29

Reply:
The core.Block ctor sets it to empty string; that might be the only time it's not a list. I guess the right thing to do is change the ctor to initialize it as empty list and don't even check for type here?